src/environmentd/src/lib.rs

```diff
@@ -360,11 +399,24 @@ pub async fn serve(config: Config) -> Result<Server, anyhow::Error> {
     if let (Some(segment_client), Some(frontegg)) = (segment_client, config.frontegg) {
         telemetry::start_reporting(telemetry::Config {
             segment_client,
-            adapter_client,
+            adapter_client: adapter_client.clone(),
             organization_id: frontegg.tenant_id(),
         });
     }
 
+    // If system_parameter_frontend is present, start the system_parameter_sync loop.
+    if let Some(system_parameter_frontend) = system_parameter_frontend {
+        let system_parameter_backend = SystemParameterBackend::new(adapter_client).await?;
+        task::spawn(
```

Comment:
I wonder if there is a way to ensure that we don't kill `environmentd` if the `system_parameter_sync` loop throws a panic. I think logging a `Sentry` error is probably and letting only the sync task die is the more robust solution here.

Reply:
As @doy-materialize mentioned on Slack, `mz_ore::panic::catch_unwind` is the way! There's an async variant too.